src/main/java/com/scalar/db/storage/multistorage/MultiStorage.java

```diff
@@ -0,0 +1,153 @@
+package com.scalar.db.storage.multistorage;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scan;
+import com.scalar.db.api.Scanner;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.service.StorageModule;
+import com.scalar.db.storage.common.util.Utility;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+
+/**
+ * A storage implementation with multi-storage for {@link DistributedStorage}.
+ *
+ * @author Toshihiro Suzuki
+ */
+@ThreadSafe
+public class MultiStorage implements DistributedStorage {
+
+  private final Map<String, DistributedStorage> storageMap;
+  private final DistributedStorage defaultStorage;
+
+  private Optional<String> namespace;
+  private Optional<String> tableName;
+
+  @Inject
+  public MultiStorage(MultiStorageDatabaseConfig config) {
+    Map<String, DistributedStorage> databaseStorageMap = new HashMap<>();
```

Comment:
I think `databaseStorageMap` is a bit confusing and not necessarily correct since it's a map of <alias-of-storage, storage-object>?
So should it be something like `aliasStorageMap` or `nameStorageMap`?